pkg/cloud/azure/services/virtualmachines/virtualmachines.go

```diff
@@ -191,11 +191,8 @@ func (s *Service) CreateOrUpdate(ctx context.Context, spec azure.Spec) error {
 		return errors.Wrapf(err, "cannot create vm")
 	}
 
-	err = future.WaitForCompletionRef(ctx, s.Client.Client)
-	if err != nil {
-		return errors.Wrapf(err, "cannot get the vm create or update future response")
-	}
-
+	// Do not wait until the operation completes. Just check the result
+	// so the call to Create actuator operation is async.
 	_, err = future.Result(s.Client)
```

Comment:
I am not sure if this actually returns an error if on creating/updating. I would rather change this to 


<Get> -> if succeeded -> return success else CreateOrUpdate -> Creating/Updating -> error

Reply:
From `github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute/models.go:10286`:

```
// Result returns the result of the asynchronous operation.
// If the operation has not completed it will return an error.
func (future *VirtualMachinesDeleteFuture) Result(client VirtualMachinesClient) (ar autorest.Response, err error) {
```

From `vendor/github.com/Azure/go-autorest/autorest/azure/async.go:171`:

```
// WaitForCo mpletion will return when one of the following conditions is met: the long
// running operation has completed, the provided context is cancelled, or the client's
// polling duration has been exceeded.  It will retry failed polling attempts based on
// the retry value defined in the client up to the maximum retry attempts.
// Deprecated: Please use WaitForCompletionRef() instead.
func (f Future) WaitForCompletion(ctx context.Context, client autorest.Client) error {
```

So in case the operation is not done, `WaitForCompletion` generates an error. The same holds for `Result`. Only when the operation is done, nil error is returned in both invocations. So `WaitForCompletion` just waits until the operation is done. Effectly making `Results` a no-op if `ar` return value of `Results` is not processed after.